types.ts

```diff
@@ -187,95 +234,202 @@ interface BaseAuthentication {
- * https://foo.com/apis/dosomething?param1=<param1 value>&param2=<param2 value>
+ * Authenticate using multiple tokens, each passed as a different URL parameter, e.g.
+ * https://example.com/api?param1=token1&param2=token2
+ *
+ * The parameter names are defined in the {@link params} array property.
  */
 export interface MultiQueryParamTokenAuthentication extends BaseAuthentication {
   type: AuthenticationType.MultiQueryParamToken;
+  /**
+   * Names and descriptions of the query parameters used for authentication.
+   */
   params: Array<{
+    /**
+     * The actual query parameter, e.g. "foo" if a token is passed as "foo=bar".
+     */
     name: string;
+    /**
+     * A description for the user of what value they should provide this parameter.
+     */
     description: string;
   }>;
 }
 
+/**
+ * Authenticate using OAuth2. You must specify the authorization URL, token exchange URL, and
+ * scopes here as part of the pack definition. You'll provide the application's client ID and
+ * client secret in the pack management UI, so that these can be stored securely.
+ *
+ * The API must use a (largely) standards-compliant implementation of OAuth2.
+ */
 export interface OAuth2Authentication extends BaseAuthentication {
   type: AuthenticationType.OAuth2;
+  /**
+   * The URL to which the user will be redirected in order to authorize this pack.
+   * This is typically just a base url with no parameters. Coda will append the `scope`
+   * parameter automatically. If the authorization flow requires additional parameters,
+   * they may be specified using {@link additionalParams}.
+   */
   authorizationUrl: string;
+  /**
+   * The URL that Coda will hit in order to exchange the temporary token for an access token
+   * at the end of the OAuth handshaek flow.
+   */
   tokenUrl: string;
+  /**
+   * Scopes that are required to use this pack.
+   */
   scopes?: string[];
+  /**
+   * A custom prefix to be used when passing the access token in the HTTP Authorization
+   * header when making requests. Typically this prefix is `Bearer` which is what will be
+   * used if this value is omitted. However, some services require a different prefix.
+   * When sending authenticated requests, a HTTP header of the form
+   * `Authorization: <tokenPrefix> <token>` will be used.
+   */
   tokenPrefix?: string;
+  /**
+   * Option custom URL parameters and values that should be included when redirecting the
+   * user to the {@link authorizationUrl}.
+   */
   additionalParams?: {[key: string]: any};
 
-  // Some OAuth providers will return the API domain with the OAuth response.
-  // This is the key in the OAuth response json body that points to the endpoint.
+  /**
+   * In rare cases, OAuth providers will return the specific API endpoint domain for the user as
+   * part of the OAuth token exchange response. If so, this is the property in the OAuth
+   * token exchange response JSON body that points to the endpoint.
```

Comment:
Is this value later accessed via `context.endpoint`? If so, let's mention that and link to that property.